---
title: "Synaptic database quering examples"
author: "Oksana Sorokina"
date: '`r format(Sys.time(), "%d.%m.%Y")`'
output:
  pdf_document:
    keep_tex: yes
    number_sections: yes
  prettydoc::html_pretty:
    theme: architect
    highlight: github
header-includes:
- \usepackage[T2A]{fontenc}
- \usepackage[utf8]{inputenc}
- \usepackage[english,russian]{babel}
- \usepackage{grffile}
- \usepackage{rotating}
- \usepackage{caption}
- \usepackage{longtable}
- \usepackage{lscape}
---
```{r loadPackages, include=FALSE, cache=FALSE}
## load additional packages in this chunk
library(pander)
library(knitr)
library(ggplot2)
library(dplyr)
library(igraph)
library(UpSetR)
library(igraph)
#dbname<-"/Users/lptolik/Desktop/synaptic.proteome.db.sqlite"
dbname<-"synaptic.proteome_SR_20210408.db.sqlite"

```
```{r functions, include=FALSE}

## Custom functions used in the analysis should go into this chunk.
## They will be listed in their own section of the appendix.
#' Function to create UpSetR compatible table from taxonomy table
#'
#' @param data taxonomy data 
#' @param cname name of the column with taxon of interest
#' @param ename name of the column with entity names
#'
#' @return UpSetR compatible data.frame
prepareUpSet<-function(data,cname,ename='GeneID'){
data<-as.data.frame(data)
myFromList<-function (input) 
{
  elements <- unique(unlist(input))
  data <- unlist(lapply(input, function(x) {
    x <- as.vector(match(elements, x))
  }))
  data[is.na(data)] <- as.integer(0)
  data[data != 0] <- as.integer(1)
  data <- data.frame(matrix(data, ncol = length(input), byrow = F))
  rownames(data)<-elements
  data <- data[which(rowSums(data) != 0), ]
  names(data) <- names(input)
  return(data)
}
  if(any(is.na(data[,cname]))){
    data[is.na(data[,cname]),cname]<-'Unspecified'
  }
  sets<-unique(data[,cname])
  l<-lapply(sets, function(.x){ 
   as.character(data[data[,cname]==.x,ename])
  })
  names(l)<-sets
  pt<-myFromList(l)
}

```

```{r setup, include=FALSE, cache=FALSE}
## This chunk should contain global configuration commands.
## Use this to set knitr options and related things. Everything
## in this chunk will be included in an appendix to document the
## configuration used.
#output <- opts_knit$get("rmarkdown.pandoc.to")
opts_knit$set(stop_on_error = 2L)


## Cache options
opts_chunk$set(cache=FALSE)


## Set 'hide.fig.code' to FALSE to include code chunks that
## produce Figures in the output. Note that this affects all chunks
## that provide a figure caption.
opts_chunk$set(hold=TRUE, hide.fig.code=FALSE)

## Pander options
panderOptions("digits", 3)
panderOptions("table.split.table", 160)

```
# Connect DB

```{r connect}
con <- DBI::dbConnect(RSQLite::SQLite(), dbname = dbname)

```
# Query
## Get specific gene information

```{r get.specific.genes}
gns<-con %>% tbl("FullGeneFullPaperFullRegion") %>% filter(HumanEntrez == "1742") %>%
  select(Localisation, HumanEntrez, HumanName, PaperPMID, Paper, Year, BrainRegion)

gns %>% show_query()
```

```{r gene.sp.tbl}
gns %>% collect() %>% pander()
```
## Get paper information for gene list

```{r get.gene.list}
gns<-con %>% tbl("FullGeneFullPaperFullRegion") %>% filter(HumanEntrez %in% c("4130","10458", "57554")) %>%
  select(Localisation, HumanEntrez, HumanName, PaperPMID, Paper, Year, BrainRegion)
gns %>% collect() %>% pander()
gns %>% show_query()
```

## Get disease information for gene list


```{r get.gene.disease.list,cache=TRUE}
gns<-con %>% tbl("FullGeneFullDisease") %>% filter(HumanEntrez %in% c("4130","10458", "57554")) %>%
  select(HumanName, HDOID, Description)
gns %>% collect() %>% pander()
gns %>% show_query()
write.table(gns, file = "SomeDisease.txt", sep = "\t", row.names = F)
```

## Get SynGO annotation for gene list

[test](http://google.com)

```{r get.gene.syngo.list,cache=TRUE}
gns<-con %>% tbl("FullGeneFullPaper") %>% filter(HumanEntrez %in% c("1742","10458", "57554")) %>%
  select(Localisation, HumanEntrez,HumanName, SynGO)
gns %>% collect() %>% unique() %>%
  mutate(SynGO = paste0("[",SynGO,"](https://syngoportal.org/gene.html?",SynGO,")")) %>% 
  pander()
gns %>% show_query()
write.table(gns, file = "SomeSynGO.txt", sep = "\t", row.names = F)
```

## Get number of papers for each gene

```{r make.tbl}
gns<-con %>% tbl("Gene") %>% filter( !is.na(MGI))
paps<- con %>% tbl("Paper")
gps<- con %>% tbl("PaperGene")
t<- gns %>% 
  inner_join(gps,by=c("ID"="GeneID")) %>% 
  inner_join(paps,by=c("PaperPMID"="PMID"))

t %>% show_query()
```

```{r aggregate.tbl}
t %>% group_by(MGI,MouseName) %>%
  summarise(numP=n()) %>% 
  arrange(desc(numP)) %>% 
  head(25) %>% 
  collect() %>%
 pander()
```
## Make the full graph

```{r make.the.full.graph}
gns<-con %>% tbl("Gene") %>% filter( !is.na(HumanEntrez)) %>% collect()
pps <- con %>% tbl("PPI") %>% select(A,B) %>%collect()
g <- graph_from_data_frame(pps, directed = FALSE, vertices = gns)
summary(g)
gr <- simplify(g)
summary(gr)
```

## Make graph from specific paper or localisation

````{r make.graph.for.specific.paper}
gnp<-con %>% tbl("FullGeneFullPaper") %>% filter(PaperPMID == "10818142") %>% select(GeneID, HumanEntrez, HumanName)
#gnp<-con %>% tbl("FullGeneFullPaper") %>% filter(Localisation == "Presynaptic") %>% select(GeneID, HumanEntrez, HumanName)
ppi <- con %>% tbl("PPI")
edges<- gnp %>% 
  inner_join(ppi,by=c("GeneID"="A")) %>% 
  inner_join(gnp,by=c("B"="GeneID")) %>% select(GeneID,B) %>% collect()
vert <- gnp %>% collect()

gp <- graph_from_data_frame(edges, directed = FALSE, vertices = vert)
gp <- simplify(gp)
summary(gp) 
coords <- layout_(gp, as_star())
plot(gp, layout = coords, vertex.label = V(gp)$HumanName)
```

# Appendix {.tabset}
## Functions
```{r functions, eval=FALSE, include=TRUE}
```

## Setup R
```{r setup, eval=FALSE}
```

## Versions

### Session Info
```{r sessionInfo, echo=FALSE, results='asis', class='text', warning=FALSE}
c<-devtools::session_info()
pander(t(data.frame(c(c$platform))))
pander(as.data.frame(c$packages)[,-c(4,5,10,11)])
```
